### ISSUE TITLE
[QST]: Building from source errors

### ISSUE THREAD

**Issue by yigithanyigit** [rapidsai/cugraph#4763]
### What is your question?

Hello,
my problem is not a question neither bug(I guess?). I am having trouble while building from source via WSL2 or inside devcontainer (24.12-cpp-gcc11-cuda12.6-ubuntu22.04). I am not sure what is the problem but I am getting consequtive errors same like this. It's not happening for every source file.

At first I tried to build like in the documentation and I got same error after that I tried to build inside the devcontainer with conda and I am still getting errors.

Thanks!

gcc version:
`11-4-0`

My nvidia-smi:

```
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 560.35.02              Driver Version: 560.94         CUDA Version: 12.6     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 3090        On  |   00000000:07:00.0  On |                  N/A |
|  0%   42C    P8             35W /  390W |    3213MiB /  24576MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A        33      G   /Xwayland                                   N/A      |
|    0   N/A  N/A        66      G   /Xwayland                                   N/A      |
+-----------------------------------------------------------------------------------------+
```

Error Message

```
[11/509] Building CUDA object CMakeFiles/cugraph.dir/src/sampling/random_walks_mg_v32_e32.cu.o
FAILED: CMakeFiles/cugraph.dir/src/sampling/random_walks_mg_v32_e32.cu.o
/usr/bin/sccache /root/miniconda3/envs/cugraph_dev/bin/nvcc -forward-unknown-to-host-compiler -ccbin=/root/miniconda3/envs/cugraph_dev/bin/g++ -DCUDA_API_PER_THREAD_DEFAULT_STREAM -DCUTLASS_NAMESPACE=raft_cutlass -DFMT_SHARED -DLIBCUDACXX_ENABLE_EXPERIMENTAL_MEMORY_RESOURCE -DRAFT_COMPILED -DRAFT_SYSTEM_LITTLE_ENDIAN=1 -DSPDLOG_FMT_EXTERNAL -DTHRUST_DEVICE_SYSTEM=THRUST_DEVICE_SYSTEM_CUDA -DTHRUST_DISABLE_ABI_NAMESPACE -DTHRUST_HOST_SYSTEM=THRUST_HOST_SYSTEM_CPP -DTHRUST_IGNORE_ABI_NAMESPACE_ERROR -Dcugraph_EXPORTS -I/root/cugraph/cpp/../thirdparty -I/root/cugraph/cpp/src -I/root/cugraph/cpp/include -I/root/cugraph/cpp/build/_deps/cccl-src/thrust/thrust/cmake/../.. -I/root/cugraph/cpp/build/_deps/cccl-src/libcudacxx/lib/cmake/libcudacxx/../../../include -I/root/cugraph/cpp/build/_deps/cccl-src/cub/cub/cmake/../.. -I/root/cugraph/cpp/build/_deps/cuco-src/include -isystem /root/miniconda3/envs/cugraph_dev/include -isystem /root/miniconda3/envs/cugraph_dev/targets/x86_64-linux/include -g -std=c++17 "--generate-code=arch=compute_86,code=[sm_86]" -Xcompiler=-fPIC --expt-extended-lambda --expt-relaxed-constexpr -Werror=cross-execution-space-call -Wno-deprecated-declarations -DRAFT_HIDE_DEPRECATION_WARNINGS -Xptxas=--disable-warnings -Xcompiler=-Wall,-Wno-error=sign-compare,-Wno-error=unused-but-set-variable -Xfatbin=-compress-all -G -Xcompiler=-rdynamic -MD -MT CMakeFiles/cugraph.dir/src/sampling/random_walks_mg_v32_e32.cu.o -MF CMakeFiles/cugraph.dir/src/sampling/random_walks_mg_v32_e32.cu.o.d -x cu -c /root/cugraph/cpp/src/sampling/random_walks_mg_v32_e32.cu -o CMakeFiles/cugraph.dir/src/sampling/random_walks_mg_v32_e32.cu.o
nvcc warning : incompatible redefinition for option 'compiler-bindir', the last value of this option was used
/root/cugraph/cpp/src/prims/detail/sample_and_compute_local_nbr_indices.cuh(600): error: name followed by "::" must be a class or namespace name
   cugraph::invalid_edge_id_v<edge_t> == cugraph::ops::graph::INVALID_ID<edge_t>
                                                  ^

/root/cugraph/cpp/src/prims/detail/sample_and_compute_local_nbr_indices.cuh(600): error: expected an expression
   ) ? static_cast<void> (0) : __assert_fail (
   ^

/root/cugraph/cpp/src/prims/detail/sample_and_compute_local_nbr_indices.cuh(1572): error: name followed by "::" must be a class or namespace name
   cugraph::invalid_edge_id_v<edge_t> == cugraph::ops::graph::INVALID_ID<edge_t>
                                                  ^

/root/cugraph/cpp/src/prims/detail/sample_and_compute_local_nbr_indices.cuh(1572): error: expected an expression
   ) ? static_cast<void> (0) : __assert_fail (
   ^

4 errors detected in the compilation of "/root/cugraph/cpp/src/sampling/random_walks_mg_v32_e32.cu".
```

### Code of Conduct

- [x] I agree to follow cuGraph's Code of Conduct
- [x] I have searched the [open issues](https://github.com/rapidsai/cugraph/issues) and have found no duplicates for this question

**Comment by ChuckHastings** [comment]
This is a bug.  Thanks for alerting us... even if you didn't realize it was a bug.

We deprecated some functionality so that we could eliminate our dependency on an internal package called cugraph-ops.  This must have been missed.

It appears that this is only referenced in two asserts in this file.  If you simply delete the two lines that perform the asserts, you probably will be able to compile.  Please let us know if that resolves your build problem.

In the meantime, I will add this issue to our items to fix before the 24.12 release.

**Comment by yigithanyigit** [comment]
First of all sorry for late answer and thank you a lot about your fast response. Normally I would like to make a patch. However, even I saw your comment I had't have time to make a patch and test it. 

I am trying to build with the changes that you suggested and it looks pretty fine right now!

Thanks!
